### PR TITLE
No element <listener> be added when enabling the telemetry with AI in Eclipse #2150

### PR DESCRIPTION
**Pull request by zmssp**
- Fix issue https://github.com/Microsoft/azure-tools-for-java/issues/2150 and https://github.com/Microsoft/azure-tools-for-java/issues/2789
- Add work round for application insight sdk. Add a listener in web.xml

### CODE REVIEW

**Comment by Eskibear** [comment]
Please refine the title
Also adding @jdneo as reviewer

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/applicationinsights/AILibraryHandler.java [hunk 1]
```diff
@@ -224,6 +225,30 @@ public void removeAIFilterDef() throws Exception {
 		}
 	}
 
+	/**
+     * WorkRound for issue https://github.com/Microsoft/azure-tools-for-java/issues/2122
```
Comment (jdneo) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/applicationinsights/AILibraryHandler.java]: workaround

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/applicationinsights/AILibraryHandler.java [hunk 2]
```diff
@@ -224,6 +225,30 @@ public void removeAIFilterDef() throws Exception {
 		}
 	}
 
+	/**
```
Comment (jdneo) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/applicationinsights/AILibraryHandler.java]: Wired indention.

File: Utils/azuretools-core/src/com/microsoft/azuretools/utils/TelemetryUtils.java [hunk 3]
```diff
@@ -25,4 +41,52 @@ public static String getMachieId(String dataFile, String prefVal, String instId)
...
+		}
+		final XPath xpath = XPathFactory.newInstance().newXPath();
+		final String exprListener = EXPR_LISTENER;
+		final Element eleListener = (Element) xpath.evaluate(exprListener, webXMLDoc, XPathConstants.NODE);
+		if (eleListener != null) {
+			eleListener.getParentNode().removeChild(eleListener);
+		}
+	}
+
+
```
Comment (jdneo) [inline Utils/azuretools-core/src/com/microsoft/azuretools/utils/TelemetryUtils.java]: redundant empty lines

Comment (zmssp) [reply Utils/azuretools-core/src/com/microsoft/azuretools/utils/TelemetryUtils.java]: done

**Pull request merged**